python/ray/data/_internal/table_block.py

```diff
@@ -27,6 +27,8 @@ class TableBlockBuilder(BlockBuilder[T]):
     def __init__(self, block_type):
         # The set of uncompacted Python values buffered.
         self._columns = collections.defaultdict(list)
+        # The sorted columns names of uncompacted Python values buffered.
+        self._sorted_columns_names = None
```

Comment:
Save columns names here because we can clear the buffer `self_columns` in `_compact_if_needed()`.